packages/jest-transform/src/types.ts

```diff
@@ -65,7 +65,7 @@ export interface TransformOptions<OptionType = unknown>
 
 export interface SyncTransformer<OptionType = unknown> {
   canInstrument?: boolean;
-  createTransformer?: (options?: OptionType) => SyncTransformer;
+  createTransformer?: (options?: OptionType) => SyncTransformer<OptionType>;
```

Comment:
all of the type changes here should have been in #9889, but I didn't test _writing_ a transformer before merging it, just calling it from `runtime`. So snuck it in here 😀